server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java

```diff
@@ -1211,132 +1215,142 @@ protected String validateConfigurationValue(final String name, String value, fin
+     * @param name name of the configuration.
+     * @param value value to validate.
+     * @param type type of the value.
+     * @param configuration if the configuration uses Config instead of ConfigKey, the Config object; null otherwise.
+     * @return if the value is valid, returns null; if not, returns an error message.
+     */
+    protected String validateValueRange(String name, String value, Class<?> type, Config configuration) {
+        if (type.equals(Float.class)) {
+            Float val = Float.parseFloat(value);
+            if (overprovisioningFactorsForValidation.contains(name) && val <= 0f) {
+                return String.format("Value for configuration [%s] should be greater than 0.", name);
+            } else if (weightBasedParametersForValidation.contains(name) && (val < 0f || val > 1f)) {
+                return String.format("Please enter a value between 0 and 1 for the configuration parameter: [%s].", name);
             }
-            return null;
         }
 
         if (type.equals(Integer.class)) {
-            try {
-                final int val = Integer.parseInt(value);
-                if (NetworkModel.MACIdentifier.key().equalsIgnoreCase(name)) {
-                    //The value need to be between 0 to 255 because the mac generation needs a value of 8 bit
-                    //0 value is considered as disable.
-                    if(val < 0 || val > 255){
-                        throw new InvalidParameterValueException(name + " value should be between 0 and 255. 0 value will disable this feature");
-                    }
+            int val = Integer.parseInt(value);
+            if (NetworkModel.MACIdentifier.key().equalsIgnoreCase(name)) {
+                // The value needs to be between 0 to 255 because the MAC generation needs a value of 8 bits
+                // 0 is considered as disabled.
+                if (val < 0 || val > 255){
+                    return String.format("[%s] value should be between 0 and 255. 0 value will disable this feature.", name);
                 }
-
-                if (UnmanagedVMsManager.ThreadsOnMSToImportVMwareVMFiles.key().equalsIgnoreCase(name) || UnmanagedVMsManager.ThreadsOnKVMHostToImportVMwareVMFiles.key().equalsIgnoreCase(name)) {
-                    if (val > 10) {
-                        throw new InvalidParameterValueException("Please enter a value between 0 and 10 for the configuration parameter: " + name + ", -1 will disable it");
-                    }
+            }
+            if (UnmanagedVMsManager.ThreadsOnMSToImportVMwareVMFiles.key().equalsIgnoreCase(name) ||
+                    UnmanagedVMsManager.ThreadsOnKVMHostToImportVMwareVMFiles.key().equalsIgnoreCase(name)) {
+                if (val > 10) {
+                    return String.format("Please enter a value between 0 and 10 for the configuration parameter: [%s]. -1 will disable it.", name);
                 }
```

Comment:
```suggestion
                    UnmanagedVMsManager.ThreadsOnKVMHostToImportVMwareVMFiles.key().equalsIgnoreCase(name)) {
                if (val < -2 || val > 10) {
                    return String.format("Please enter a value between 0 and 10 for the configuration parameter: [%s]. -1 will disable it.", name);
                }
```

Reply:
Shouldn't the returned message be updated as well?

Reply:
Shouldn't the returned message be updated as well?